### PR TITLE
fix(Utilities): Account for column reversed scrollable elements in allowScrollOnElement

### LINKED ISSUE #30132: [Bug]: using "flex-direction: column-reverse" scroll is not working in iOS and Andriod mobile phones
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 11 10.0.22621
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
    Memory: 15.69 GB / 31.73 GB
  Browsers:
    Edge: Chromium (120.0.2210.77)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

## Actual Behavior
When the content reaches the height more than Panel Scrollable content, user is not able to scroll to up to view the content.
and user has to try scrolling multiple times up/down, to make it work.

## Expected Behavior
Panel Content should be scrollable smoothly.

### Logs

```shell

Affected Fluent UI control -> Panel.

```

### Requested priority

High

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by sopranopillow**
## Previous Behavior

Setting allowScrollOnElement for an element that has `flex-direction: column-reversed` would cause scroll issues since it didn't account for scrollTop to start from the bottom rather than the top of the scrollHeight. This would then prevent scrolling up since the change in scroll would be negative rather than positive.

Note: This affects only mobile devices and there's a hack to get around it without this fix, `allowTouchBodyScroll: true`. This is not a desirable fix since now scrolling through allows body scroll causing a refresh when scrolling past the scrollHeight.

## New Behavior

allowScrollOnElement now checks if the element is reversed to account for scrollTop starting at the bottom of the scrollHeight.

## Related Issue(s)

- Fixes #30132

### CODE REVIEW

File: packages/react-examples/src/react/Panel/Panel.Basic.Example.tsx [hunk 1]
```diff
@@ -15,8 +15,19 @@ export const PanelBasicExample: React.FunctionComponent = () => {
...
+            gap: '20px',
+          },
+        }}
       >
-        <p>Content goes here.</p>
+        {Array(100)
+          .fill(0)
+          .map((_, index) => (
+            <DefaultButton key={index}>{index}</DefaultButton>
+          ))}
```
Comment (sopranopillow) [inline packages/react-examples/src/react/Panel/Panel.Basic.Example.tsx:30]: Remove this before merging

File: packages/utilities/src/scroll.ts [hunk 2]
```diff
@@ -57,19 +58,26 @@ export const allowScrollOnElement = (element: HTMLElement | null, events: EventG
...
+      computedStyles = getComputedStyle(_element);
     }
 
+    const scrollTop = _element.scrollTop;
+    const isColumnReverse = computedStyles?.flexDirection === 'column-reverse';
+
     // if the element is scrolled to the top,
     // prevent the user from scrolling up
-    if (_element.scrollTop === 0 && clientY > 0) {
+    if (scrollTop === 0 && (isColumnReverse ? clientY < 0 : clientY > 0)) {
```
Comment (sopranopillow) [inline packages/utilities/src/scroll.ts:71]: this is the main culprit of the scroll issue

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/167cccee21a690dd231dd3b4d4d5159cfd8169e2'>167cccee21a690dd231dd3b4d4d5159cfd8169e2</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/utilities/src/scroll.ts [hunk 3]
```diff
@@ -24,12 +24,15 @@ export const DATA_IS_SCROLLABLE_ATTRIBUTE = 'data-is-scrollable';
...
-  if (!element) {
+  const window = getWindow(element);
+
+  if (!element || !window) {
     return;
   }
 
   let _previousClientY = 0;
   let _element: Element | null = null;
+  let computedStyles: undefined | CSSStyleDeclaration = window.getComputedStyle(element);
```
Comment (tomi-msft) [inline packages/utilities/src/scroll.ts:35]: nit: switch the order of the types

**Review by tomi-msft — APPROVED** [review]

**Pull request merged**